### PR TITLE
Add CSCS GitLab CI config

### PR DESCRIPTION
**Pull request by msimberg**
Adds a dummy config. This is work in progress.

### CODE REVIEW

File: .cscs/ci.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
+include:
```
Comment (hkaiser) [inline .cscs/ci.yml:1]: Do you plan to enforce the same requirements to the new repository as we've had on the old one? If yes, then please add copyright and license information.

Comment (msimberg) [reply .cscs/ci.yml:1]: You're fast! Yes, definitely :) This config doesn't even do anything HPX-related yet, and will change still significantly. Once it's working I'll add the same to HPX as well. But I suspect it's going to take a while given that the underlying system is still a work in progress.

However, in the meantime, could you check if this loads for you: https://gitlab.com/cscs-ci/ci-testing/webhook-ci/hpx-local-4750890491460239/-/pipelines/432028314? If yes, the logs will be public once this is working.

Comment (msimberg) [reply .cscs/ci.yml:7]: I just wanted to add that inspect will still check for these things like on STEllAR-GROUP/hpx.

**Comment by msimberg** [comment]
@hkaiser I'm closing this due to https://github.com/STEllAR-GROUP/hpx/pull/5734.

Just FYI, this is long term meant to be a replacement for the mirrored repo https://gitlab.com/cscs-ci/STEllAR-GROUP/hpx-ci/. It's still based on GitLab CI but the mirroring is done automatically at CSCS. If you're interested in using it in the future I can let you know when things are running and a bit more stable (the middleware is still a work in progress).

**Pull request closed without merging**